tox.ini

```diff
@@ -61,14 +58,14 @@ commands = black {posargs} qiskit_experiments test tools setup.py
 [testenv:docs]
 passenv = EXPERIMENTS_DEV_DOCS
 commands =
-  sphinx-build -b html {posargs} docs/ docs/_build/html
+  sphinx-build -j auto -T --keep-going -b html {posargs} docs/ docs/_build/html
```

Comment:
I think `-j auto` is okay but we might want to check on Windows. I don't think Sphinx supports parallel builds on Windows because it relies on process forking. I think it quietly falls back to serial though.

Reply:
we do `-j auto` on Terra, but I don't know of anybody building the docs on windows, so dunno if there are problems.

`--keep-going` is unimportant unless you also set `-W` (treat warnings as errors) as well

Reply:
`-j auto` will likely prove unreliable if `jupyter-execute` is being used. There are a lot of potential race conditions in the jupyter sphinx extension around launching the kernels and every time I've tried to combine it with parallel builds I've had issues.

In general I would suggest enabling `-W` most sphinx warnings typically point to real bugs in the documentation.

Reply:
We haven't had problems with `-j auto` so far, it just doesn't parallelize the `jupyter-execute` work while saving time on the rest of the build. If we start getting race conditions we'll disable it. The plan is to turn on `-W` once we've gotten rid of all the Sphinx warnings.

Reply:
#1070 was just merged, so all the sphinx warnings should be gone now after merging main. For now, we are putting `-W` in the GitHub workflow rather than in tox.ini because there are extra sphinx warnings that only show up on Mac. Maybe `--keep-going` should go in the workflow as well?

I was worried about combining sphinx parallelization and jupyter-sphinx because sphinx uses the multiprocessing fork context and qiskit-experiments has some global state that would not combine well with it (a ThreadPoolExecutor in a class variable), but it did not seem to be a problem in practice. It seemed like jupyter-sphinx was launching a new subprocess for each rst file (I added something to print out the PID and was seeing a different one for each notebook).

Reply:
I turned off `-j auto` for the default `tox -edocs` command because it was causing crashes on Mac (it crashes randomly on my Mac without multiprocessing too but is more frequent with it on). Since we haven't had an issue on Linux yet, I'm keeping it for the CI build.